tests/test_unit/test_visualization.py

```diff
@@ -0,0 +1,172 @@
+from unittest.mock import MagicMock, patch
+
+import numpy as np
+import pytest
+import torch
+
+from crabs.detection_tracking.visualization import (
+    draw_bbox,
+    draw_detection,
+    save_images_with_boxes,
+)
+
+
+@pytest.fixture
+def sample_image():
+    # Create a sample image for testing
+    return np.zeros((100, 100, 3), dtype=np.uint8)
+
+
+@pytest.mark.parametrize(
+    "top_left, bottom_right, color, label_text",
+    [
+        ((10, 10), (50, 50), (0, 255, 0), None),
+        ((10, 10), (50, 50), (0, 255, 0), "Test Label"),
+    ],
+)
+def test_draw_bbox(sample_image, top_left, bottom_right, color, label_text):
```

Comment:
this test seems to test slightly the wrong thing:
- If a text is passed, it checks the text was drawn (but it doesn't check the box was)
- If no text is passed, then it checks the box was drawn.

I think I would either:
- Option 1: modify this test so that the box is always checked (regardless of whether a text is passed or not).
- Option 2: keep only the text test (I think it was like this in the previous review). For the box, you already have the `test_draw_bbox_crop`, which is stricter than the check you do here. 

I think Option 2 is best. For the text test, this is what you had before:

```python
def test_draw_bbox_with_label(sample_image):
    top_left = (10, 10)
    bottom_right = (50, 50)
    color = (0, 255, 0)
    label_text = "Test Label"

    draw_bbox(sample_image, top_left, bottom_right, color, label_text)

    # Check if label text is drawn
    assert np.any(sample_image[10:20, 10:100] == color)
```

Small improvement: can we link the coordinates where we expect the text to be (last line in the snippet above) to the bounding box coordinates?